units/crons.go

```diff
@@ -323,51 +326,89 @@ func PopulateSchedulerJobs() amboy.QueueOperation {
 			"op":       "dispatcher",
 		})
 
+		grip.Error(message.WrapError(err, message.Fields{
+			"cron":      schedulerJobName,
+			"impact":    "new task scheduling non-operative",
+			"operation": "background task creation",
+		}))
+
 		ts := util.RoundPartOfMinute(20)
 		for _, id := range names {
 			lastRun, ok := lastPlanned[id]
 			if ok && time.Since(lastRun) < 40*time.Second {
 				continue
 			}
 
-			job := NewDistroSchedulerJob(id, ts)
-			catcher.Add(queue.Put(job))
+			catcher.Add(queue.Put(NewDistroSchedulerJob(id, ts)))
 		}
 
 		return catcher.Resolve()
 	}
 }
 
-// used for infrequently running system alerts
-func PopulateAlertingJobs() amboy.QueueOperation {
+// PopulateHostAlertJobs adds alerting tasks infrequently for host
+// utilization monitoring.
+func PopulateHostAlertJobs() amboy.QueueOperation {
 	return func(queue amboy.Queue) error {
 		catcher := grip.NewBasicCatcher()
-		catcher.Add(addHostLongRunningTaskJobs(queue))
-		catcher.Add(addHostStatsJob(queue))
+
+		ts := util.RoundPartOfHour(20).Format(tsFormat)
+
+		hosts, err := host.Find(host.IsRunningTask)
+		grip.Warning(message.WrapError(err, message.Fields{
+			"cron":      hostAlertingName,
+			"operation": "background task creation",
+			"impact":    "admin alerts not set",
+		}))
+
+		catcher.Add(err)
+		if err == nil {
+			for _, host := range hosts {
+				catcher.Add(queue.Put(NewHostAlertingJob(host, ts)))
+			}
+		}
+
+		catcher.Add(queue.Put(NewHostStatsJob(ts)))
 
 		return catcher.Resolve()
 	}
 }
 
-func addHostLongRunningTaskJobs(queue amboy.Queue) error {
-	hosts, err := host.Find(host.IsRunningTask)
-	if err != nil {
-		return errors.WithStack(err)
-	}
+func PopulateAgentDeployJobs(env evergreen.Environment) amboy.QueueOperation {
+	return func(queue amboy.Queue) error {
+		flags, err := evergreen.GetServiceFlags()
+		if err != nil {
+			return errors.WithStack(err)
+		}
 
-	ts := util.RoundPartOfHour(2).Format(tsFormat)
-	catcher := grip.NewBasicCatcher()
+		if flags.TaskrunnerDisabled {
+			grip.InfoWhen(sometimes.Percent(evergreen.DegradedLoggingPercent), message.Fields{
+				"message": "taskrunner disabled",
+				"impact":  "agents are not deployed",
+				"mode":    "degraded",
+			})
+			return nil
+		}
 
-	for _, host := range hosts {
-		job := NewHostAlertingJob(host, ts)
-		catcher.Add(queue.Put(job))
-	}
+		hosts, err := host.Find(host.NeedsNewAgent(time.Now()))
+		grip.Error(message.WrapError(err, message.Fields{
+			"operation": "background task creation",
+			"cron":      agentDeployJobName,
+			"impact":    "agents cannot start",
```

Comment:
This is actually a database error, I think.

Reply:
true, but the impact is the same, added some clarification